File: prob1.java
// Time Complexity : O (n) 
// Space Complexity : O (H)
// Did this code successfully run on Leetcode : Yes
// Any problem you faced while coding this : while swap, the swap without temp variable failed
// in cases where the indices were same

// can be dine by bfs also

import java.util.*;
public class prob1 {
    public class TreeNode {
     int val;
     TreeNode left;
     TreeNode right;
     TreeNode() {}
     TreeNode(int val) { this.val = val; }
     TreeNode(int val, TreeNode left, TreeNode right) {
         this.val = val;
         this.left = left;
         this.right = right;
     }
 }
    List<List<Integer>> result;
    public List<List<Integer>> levelOrder(TreeNode root) {
        result = new ArrayList<>();
        if(root == null) return result;
        dfs(root,0);
        return result;
        
    }

    public void dfs(TreeNode root, int level){
        if(root == null) return;
      
        if(level == result.size()){
            result.add(new ArrayList<>());
        }
        result.get(level).add(root.val);
        dfs(root.left, level + 1);
        dfs(root.right, level + 1);
    }
    
}

File: prob1.py
# Time Complexity : O(N) for while loop
# Space Complexity : O(H) for recursion stack
# Did this code successfully run on Leetcode : Yes
# Any problem you faced while coding this : No
# Your code here along with comments explaining your approach

# using dfs approach, for every level, 
# if length of result array == level, 
# that means there is no nested array for this level,
# so append a list to the result array, 
# else just add the element to the nested array in result   
# array whose index is level


class TreeNode:

    def __init__(self, val=0, left=None, right=None):
        self.val = val 
        self.left = left
        self.right = right


class Solution:

    result = []
    def levelOrder(self, root):
        global result
        result = []
        if root is None:
            return result
        self.dfs(root, 0)
        return result

    def dfs(self, root, level):
        if root is None:
            return 
        if len(result) == level:
            result.append([])
        result[level].append(root.val) 

        self.dfs(root.left, level + 1)
        self.dfs(root.right, level + 1)
    